src/argilla_server/models/database.py

```diff
@@ -357,6 +357,21 @@ def metadata_property_by_name(self, name: str) -> Union["MetadataProperty", None
             if metadata_property.name == name:
                 return metadata_property
 
+    def question_by_id(self, question_id: UUID) -> Union[Question, None]:
+        for question in self.questions:
+            if question.id == question_id:
+                return question
+
+    def question_by_name(self, name: str) -> Union["Question", None]:
+        for question in self.questions:
+            if question.name == name:
+                return question
+
+    def vector_settings_by_name(self, name: str) -> Union["VectorSettings", None]:
+        for vector_settings in self.vectors_settings:
+            if vector_settings.name == name:
+                return vector_settings
+
```

Comment:
Not sure if these methods should be defined in the model. They are only used by the bulk.